src/Umbraco.Infrastructure/Mail/EmailSender.cs

```diff
@@ -70,12 +75,54 @@ private async Task SendAsyncInternal(EmailMessage message, string emailType, boo
                 }
             }
 
-            if (_globalSettings.IsSmtpServerConfigured == false)
+            var isPickupDirectoryConfigured = !string.IsNullOrWhiteSpace(_globalSettings.Smtp?.PickupDirectoryLocation);
+
+            if (_globalSettings.IsSmtpServerConfigured == false && !isPickupDirectoryConfigured)
             {
                 _logger.LogDebug("Could not send email for {Subject}. It was not handled by a notification handler and there is no SMTP configured.", message.Subject);
                 return;
             }
 
+            if (isPickupDirectoryConfigured && !string.IsNullOrWhiteSpace(_globalSettings.Smtp?.From))
+            {
```

Comment:
We should add a link to the original implementation, in case we one day have to debug in this code
```suggestion
            {
            // The following code snippet is the recommended way to handle PickupDirectoryLocation. 
            // See more https://github.com/jstedfast/MailKit/blob/master/FAQ.md#q-how-can-i-send-email-to-a-specifiedpickupdirectory
```